aptos-move/block-executor/src/scheduler.rs

```diff
@@ -254,7 +255,12 @@ impl Scheduler {
                             // Upgrade the execution status read lock to write lock.
                             // Can commit.
                             *status_write = ExecutionStatus::Committed(incarnation);
+
                             *commit_idx += 1;
+                            if *commit_idx == self.num_txns {
+                                // All txns have been committed, the parallel execution can finish.
+                                self.done_marker.store(true, Ordering::SeqCst);
```

Comment:
we probably should remove the redundant one above

Reply:
unfortunately not redundant as is because of num_txns = 0 case. But (a) I could change to checking that or (b) even better, just don't create scheduler in that case or mark done from the beginning. I will find some way to do (b) - let me know if you disagree.